tests/unittests/test_medium_client.py

```diff
@@ -1,118 +1,174 @@
 from collections.abc import Generator
 from unittest import mock
-from unittest.mock import patch
 
-import backoff
 import requests
 import requests_mock
 from pytest import raises
+
 from tap_mixpanel import client
-from tap_mixpanel.client import (ReadTimeoutError, Server5xxError,
-                                 Server429Error, MixpanelInternalServiceError)
+from tap_mixpanel.client import (
+    MixpanelInternalServiceError,
+    ReadTimeoutError,
+    Server5xxError,
+    Server429Error,
+)
 from tests.configuration.fixtures import mixpanel_client
 
 
-@mock.patch('time.sleep', return_value=None)
+@mock.patch("time.sleep", return_value=None)
 def test_request_export_backoff_on_timeout(mock_sleep, mixpanel_client):
-    with requests_mock.Mocker() as m:
-        m.request('GET', 'http://test.com',
-                  exc=requests.exceptions.Timeout('Timeout on request'))
-
-        with raises(ReadTimeoutError) as ex:
-            for record in mixpanel_client.request_export('GET', url='http://test.com'):
+    """
+    Test that request_export method of the client backoff max times
+    (time.sleep called 'Max-1' times) if timeout error occur.
+    """
+    with requests_mock.Mocker() as mocker:
+        mocker.request(
+            "GET",
+            "http://test.com",
+            exc=requests.exceptions.Timeout("Timeout on request"),
+        )
+
+        with raises(ReadTimeoutError):
+            for record in mixpanel_client.request_export("GET", url="http://test.com"):
                 pass
+
         # Assert backoff retry count as expected
         assert mock_sleep.call_count == client.BACKOFF_MAX_TRIES_REQUEST - 1
```

Comment:
Use unittests assertions.

```suggestion
        self.assertEqual(mock_sleep.call_count, client.BACKOFF_MAX_TRIES_REQUEST - 1)
```

Reply:
In genral this test file uses built-in assertion. Please replace all those with unittests assertions.

Reply:
removed this file